scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableUnless.scala

```diff
@@ -0,0 +1,115 @@
+package scalafix.internal.rule
+
+import metaconfig.{Conf, Configured}
+
+import scala.meta._
+import scala.meta.transversers.Traverser
+import scalafix._
+import scalafix.internal.config.DisableUnlessConfig
+import scalafix.internal.util.SymbolOps
+import scalafix.lint.LintCategory
+import scalafix.rule.RuleCtx
+import scalafix.util.SymbolMatcher
+
+object DisableUnless {
+
+  /**
+    * Searches for specific nodes in Tree and adds result of searching to the list.
+    * Also passes context through tree traversal.
+    * fn can either return a value. that stops the searching, or change the context.
+    */
+  class SearcherWithContext[T, U](initContext: U)(
```

Comment:
I'm not sure the genericity is helping much here since `SearcherWithContext` is only constructed once